dlme_airflow/drivers/xml.py

```diff
@@ -0,0 +1,73 @@
+import intake
+import logging
+import pandas as pd
+import requests
+from lxml import etree
+
+
+class XmlSource(intake.source.base.DataSource):
+    container = "dataframe"
+    name = "xml"
+    version = "0.0.1"
+    partition_access = True
+
+    def __init__(self, collection_url, dtype=None, metadata=None):
+        super(XmlSource, self).__init__(metadata=metadata)
+        self.collection_url = collection_url
+        self._path_expressions = self._get_path_expressions()
+        self._records = []
+
+    def _open_collection(self):
+        collection_result = requests.get(self.collection_url).content
+        xtree = etree.fromstring(collection_result)
+        for counter, element in enumerate(xtree.findall(".//item"), start=1):
+            record = self._construct_fields(element)
+            self._records.append(record)
+
+    def _construct_fields(self, manifest: etree) -> dict:
+        output = {}
+        for field in self._path_expressions:
+            path = self._path_expressions[field]['path']
+            namespace = self._path_expressions[field]['namespace']
+            optional = self._path_expressions[field]['optional']
+            result = manifest.xpath(path, namespaces=namespace)
+            if len(result) < 1:
+                if optional is True:
+                    # Skip and continue
+                    continue
+                else:
+                    logging.warn(f"Manifest missing {field}")
+            else:
+                try:
+                    output[field] = self.sanitize_value(result[0].text)  # Use first value
```

Comment:
Maybe trying `next(result.itertext())` would strip out any html elements in the value. Noticed for example this field in csv:

```<p style=""text-align:justify;"">Episode 130: Writing on Kingdom Walls: Practices, Narratives and Visual Politics of Graffiti and Street Art in Jordan and Morocco</p>  <p style=""text-align:justify;"">Soufiane’s focus is a comparative study on cultural practices and narratives related to art production and its entanglement with resistance and visual politics in North Africa and the Middle East. By working on Morocco and Jordan, he mainly focus on wall-writings, street art, and graffiti in order to understand what wall expressions do, the extent to which they have a particularly political place in society, and how they relate to socio-political transformations.</p>  Soufiane Chinig is a first-year PhD student of anthropology in the <a href='https://www.bgsmcs.fu-berlin.de'>Berlin Graduate School Muslim Cultures and Societies</a> at <a href='https://www.fu-berlin.de/en/'>Freie Universität Berlin</a>. His research in anthropology is on writing and painting on walls in Morocco and Jordan. He also holds an MA in Sociology and Anthropology from <a href='http://www.univh2c.ma'>Hassan II University</a> in Mohammedia, and a BA in Sociology from <a href='http://www.um5.ac.ma/um5/'>Mohammed V University</a>. Alongside his academic work, he is also active in promoting Moroccan cultural heritage and evaluating urban policies in that country. This episode was recorded on July 31st 2021, at the <a href='http://legation.ipower.com/blog/'>Tangier American Legation Institute for Moroccan Studies</a> (<a href='https://www.facebook.com/Tangier-American-Legation-Institute-for-Moroccan-Studies-99255782783/'>TALIM</a>).    Posted by Hayet Lansari, Librarian, Outreach Coordinator, Content Curator (CEMA)."``` contains HTML.

Reply:
@jermnelson I think the problem we saw Friday was that the field isn't XML so the eTree `itertext()` isn't removing HTML tags. However, this got me down the path to find the lxml.html.cleaner which seems to do what we're hoping for.